metatomic-torch/include/metatomic/torch/io/zip.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace metatomic_torch {
+namespace io {
+
+/**
+ * RAII wrappers around miniz with:
+ *  - default "stored" (no compression) level
+ *  - file and in-memory (BytesIO-like) modes
+ */
+class ZipWriter {
+public:
+  ZipWriter();
+  ~ZipWriter();
+
+  ZipWriter(const ZipWriter&) = delete;
+  ZipWriter& operator=(const ZipWriter&) = delete;
+  ZipWriter(ZipWriter&& other) noexcept;
+  ZipWriter& operator=(ZipWriter&& other) noexcept;
+
+  // Open a zip archive that writes directly to a file on disk.
+  // flags are miniz writer flags (0 is fine).
+  void open_file(const std::string& path, unsigned int flags = 0);
+
+  // Open a zip archive that writes to a heap buffer (in-memory).
+  // initial_allocation_size is a hint; 0 is fine.
+  // flags are miniz writer flags (0 is fine).
+  void open_memory(size_t initial_allocation_size = 0, unsigned int flags = 0);
```

Comment:
Why are these member functions instead of constructors/static functions that behave like constructors?

Reply:
"I initially used open_*() to (a) allow default-construct + late-open + move-assign, and (b) reuse the same object for file vs memory without changing types. But a more RAII-ish API is nicer."

Reply:
Can you change it?